lxd/util/resources.go

```diff
@@ -52,6 +53,24 @@ func parseCpuinfo() ([]thread, error) {
 			t = &thread{}
 			t.ID = uint64(id)
 
+			files, err := ioutil.ReadDir(fmt.Sprintf("/sys/devices/system/cpu/cpu%d", t.ID))
+			if err != nil {
+				return nil, err
+			}
+
+			for _, file := range files {
+				if strings.HasPrefix(file.Name(), "node") {
+					nodeID := strings.TrimPrefix(file.Name(), "node")
+					nr, err := strconv.ParseUint(nodeID, 10, 64)
+					if err != nil {
+						return nil, err
+					}
+					t.numaNode = nr
+
+					break
+				}
+			}
```

Comment:
This snippet is a verbatim copy of the one above. I didn't look into these files, but also the surrounding code seems exactly the same (at least the few lines that show up in this diff). Any chance we could extract this and the rest of common code into something common?

Reply:
Eventually, yes, I'm planning on re-doing the way we do devices tracking in the near future, this will be part of it.